src/Microsoft.DotNet.ProjectModel/ProjectModelPlatformExtensions.cs

```diff
@@ -28,7 +29,8 @@ private static void CollectDependencies(IDictionary<string, LibraryExport> expor
             foreach (var dependency in dependencies)
             {
                 var export = exports[dependency.Name];
-                if (export.Library.Identity.Version.Equals(dependency.VersionRange.MinVersion))
+                if (export.Library.Identity.Version.Equals(dependency.VersionRange.MinVersion) 
+                    && !exclusionList.Contains(dependency.Name))
```

Comment:
HashSet.Add returns a bool:
true if the element is added to the HashSet<T> object; false if the element is already present.

So you shouldn't have to check Contains before Add.  You can just call "Add".  And if Add returns 'true', then walk the dependencies.